analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ProjectConfigReaderTests.cs

```diff
@@ -0,0 +1,126 @@
+            sut.OutPath.Should().Be(@"C:\foo\bar\.sonarqube\out\0");
+            sut.ProjectType.Should().Be(ProjectType.Product);
+            sut.TargetFramework.Should().Be("netcoreapp3.1");
+        }
+
+        [TestMethod]
+        public void WhenProjectConfigHasUnixPathAndTestType_ReturnsCorrectValue()
+        {
+            var options = TestHelper.CreateOptions($"ResourceTests\\SonarProjectConfig\\Valid_Test_UnixPaths\\SonarProjectConfig.xml");
+
+            var sut = new ProjectConfigReader(options);
+
+            sut.AnalysisConfigPath.Should().Be(@"/home/user/.sonarqube/conf/SonarQubeAnalysisConfig.xml");
+            sut.ProjectPath.Should().Be(@"/home/user/AwesomeBankWeb.CSharp.csproj");
+            sut.FilesToAnalyzePath.Should().Be(@"/home/user/.sonarqube/conf/0/FilesToAnalyze.txt");
+            sut.OutPath.Should().Be(@"/home/user/.sonarqube/out/0");
+            sut.ProjectType.Should().Be(ProjectType.Test);
+            sut.TargetFramework.Should().Be("net5");
+        }
+
+        [TestMethod]
+        public void WhenProjectConfigHasMixedSeparators_FilesToAnalyzePath_ReturnsValues()
+        {
+            var options = TestHelper.CreateOptions($"ResourceTests\\SonarProjectConfig\\Invalid_MixedSeparators\\SonarProjectConfig.xml");
+
+            var sut = new ProjectConfigReader(options);
+
+            sut.AnalysisConfigPath.Should().Be(@"c:\foo\bar\.sonarqube/conf/SonarQubeAnalysisConfig.xml");
+            sut.ProjectPath.Should().Be(@"C:\foo\bar\AwesomeBankWeb.CSharp/AwesomeBankWeb.CSharp.csproj");
+            sut.FilesToAnalyzePath.Should().Be(@"c:\foo\bar\.sonarqube\conf/0/FilesToAnalyze.txt");
+            sut.OutPath.Should().Be(@"C:\foo\bar\.sonarqube/out/0");
+            sut.ProjectType.Should().Be(ProjectType.Product);
+            sut.TargetFramework.Should().Be("netcoreapp3.1");
+        }
+
+        [TestMethod]
+        public void WhenProjectConfigHasMissingFilesToAnalyzePath_ReturnsCorrectValue()
+        {
+            var options = TestHelper.CreateOptions($"ResourceTests\\SonarProjectConfig\\Invalid_MissingFilesToAnalyzePath\\SonarProjectConfig.xml");
+
+            var sut = new ProjectConfigReader(options);
+
+            sut.AnalysisConfigPath.Should().Be(@"c:\foo\bar\.sonarqube\conf\SonarQubeAnalysisConfig.xml");
+            sut.ProjectPath.Should().Be(@"C:\foo\bar\AwesomeBankWeb.CSharp\AwesomeBankWeb.CSharp.csproj");
+            sut.OutPath.Should().Be(@"C:\foo\bar\.sonarqube\out\0");
+            sut.ProjectType.Should().Be(ProjectType.Product);
+            sut.TargetFramework.Should().Be("netcoreapp3.1");
+        }
+
+        [DataTestMethod]
+        [DataRow("Invalid_MissingAnalysisConfigPath")]
+        [DataRow("Invalid_MissingOutPath")]
+        [DataRow("Invalid_MissingProjectPath")]
+        [DataRow("Invalid_MissingProjectType")]
+        [DataRow("Invalid_MissingTargetFramework")]
```

Comment:
Let's rename these cases from "Invalid_MissingXxxx" to "Missing_Xxx".
The XML schema and structure is valid. It has just unexpected values for whatever reason on S4MSB side, or weird project type or structure, or some new kind of project type that's not yet supported.
Invalid are only these cases:
```
Invalid_DifferentClassName
Invalid_DifferentNamespace
Invalid_Xml
```

Reply:
We can also decide to remove something from the XML file in the future